### DISCUSSION TITLE
how to find out the DAG's start_date?

### DISCUSSION THREAD

**Discussion by jinzishuai** [apache/airflow#16293] [Q&A]
I've read a lot of documentation mentioning the importance of `start_date`. However, I don't seem to be able to find out its value for my DAG.

The UI shows None under DAG details
![image](https://user-images.githubusercontent.com/1074685/120961665-7bccdd00-c71b-11eb-91fa-c10e08d1c676.png)

However, in the dag code, I have `"start_date": airflow.utils.dates.days_ago(0)`.
So in that sense, shouldn't I see a value that is the `00:00:00.000` UTC on the same day of my DAG creation? Why does the UI show None then?

Also, I checked my postgres DB and I don't even see the `start_date` as a field in the `dag` table.

I am confused. Thanks in advance.

**Comment by josh-fell** [comment]
The `start_date` displayed is from the `serialized_dag` table in the metastore DB.  However, with that `start_date` definition for a brand new DAG, the value will be "None".  It's best practice to use a static value for `start_date` in all DAGs.  This isn't the only unexpected behavior that can come from a non-static `start_date`.

**Reply by jinzishuai** [reply]
thank you very much. I also read a bit about [dag serialization](https://airflow.apache.org/docs/apache-airflow/stable/dag-serialization.html).  My `serialized_dag` table is empty, which I guess is because I am not using static start_date? Also I am still using airflow 1.10.10.

**Reply by josh-fell** [reply]
Ah yes, with 1.10.10 serializing DAGs is not enabled by default like it is with 2.0+ hence  the empty`serialized_dag` table.  If DAGs aren't being serialized, the `start_date` comes from DAG Bag directly.  Here are pointers to the code to reference.  I suspect it might still be related to the use of the dynamic `start_date`.

**DAG Detail UI View Backend**
This retrieves the data to render in the UI.
https://github.com/apache/airflow/blob/4aec433e48dcc66c9c7b74947c499260ab6be9e9/airflow/www/views.py#L747

**DagBag Logic**
Used in the DAG Detail View Backend to retrieve the DAG information.
https://github.com/apache/airflow/blob/4aec433e48dcc66c9c7b74947c499260ab6be9e9/airflow/models/dagbag.py#L122

**DAG Detail HTML**
https://github.com/apache/airflow/blob/v1-10-stable/airflow/www/templates/airflow/dag_details.html
